vitrina/datasets/mixins.py

```diff
@@ -0,0 +1,110 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+from django.utils.translation import gettext_lazy as _
+from django.urls import reverse
+
+from vitrina.datasets.models import Dataset
+from vitrina.orgs.models import Organization
+
+
+@dataclass(frozen=True)
+class Crumb:
+    title: str
+    url: str | None
+    is_current: bool = False
+
+
+class BaseBreadcrumbsMixin:
+    include_home_in_breadcrumbs: bool = True
+    breadcrumb_title: Optional[str] = None
+    breadcrumb_url: Optional[str] = None
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["breadcrumbs"] = self.get_breadcrumbs()
+        return context
+
+    def breadcrumbs_home(self) -> List[Crumb]:
+        return [Crumb(title=_("Pradžia"), url=reverse("home"))]
+
+    def get_breadcrumb_title(self) -> str | None:
+        """Get custom breadcrumb title - override in subclasses"""
+        return self.breadcrumb_title
+
+    def get_breadcrumb_url(self) -> str | None:
+        """Get custom breadcrumb URL - override in subclasses"""
+        return self.breadcrumb_url
+
+    def append_current_crumb(self, crumbs: List[Crumb]) -> List[Crumb]:
+        """Add current page crumb to breadcrumbs"""
+        title = self.get_breadcrumb_title()
+        if title:
+            crumbs.append(Crumb(title=title, url=self.get_breadcrumb_url(), is_current=True))
+        return crumbs
+
+    def get_breadcrumbs(self) -> List[Crumb]:
+        """Default breadcrumb generation - override in subclasses"""
+        if self.include_home_in_breadcrumbs:
+            return self.breadcrumbs_home()
+        return []
+
+    def breadcrumbs_organization(self, organization: Organization) -> List[Crumb]:
+        """Generate breadcrumbs up to organization level"""
+        crumbs = []
+        if self.include_home_in_breadcrumbs:
+            crumbs.extend(self.breadcrumbs_home())
+        crumbs.append(Crumb(title=organization.title, url=organization.get_absolute_url()))
+        return crumbs
+
+
+class DatasetBreadcrumbsMixin(BaseBreadcrumbsMixin):
+    def _org_for_dataset(self, dataset: Dataset) -> Organization:
+        ancestors = dataset.get_ancestors()
+        if ancestors.exists():
+            return ancestors.first().organization
```

Comment:
```suggestion
        if ancestor := ancestors.first():
            return ancestor.organization
```

will you save one hit to DB.

Reply:
Changed to suggestion